task_processing/executors/task_executor.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import abc
+import uuid
+from collections import namedtuple
+
+import six
+
+TaskConfig = namedtuple(
+    'TaskConfig',
+    ['task_id', 'name', 'image', 'cmd', 'cpus', 'mem', 'disk', 'volumes',
+     'ports', 'cap_add', 'ulimit', 'docker_parameters'],
+)
+
+
+def make_task_config(task_id=None, name='foo', image="ubuntu:xenial",
```

Comment:
I'd like to avoid this anti-pattern I've seen in PaaSTA where we want defaults so we make methods and wrapper classes everywhere. What if we just used something off the shelf like attrs (https://github.com/python-attrs/attrs)?